app/src/main/java/com/github/se/signify/ui/Utils.kt

```diff
@@ -454,65 +454,72 @@ fun InfoPopup(onDismiss: () -> Unit, helpTitle: String, helpText: String) {
 /**
  * A reusable composable function that creates a row for statistics.
  *
- * Important to note that the list have the same size and the inner list should match their sizes.
- * Example (the number are the size): list1 = ((2), (1), (2)) => list2 = ((2), (1), (2)) and list3 =
- * (3)
+ * Important to note that the lists must have the same size.
  *
- * @param rowTestTag The principal tag for the row.
+ * @param columnTestTag The principal tag for the column.
+ * @param rowTestTag The principal tag for the row of stats.
  * @param lineText The text description for the statistic to show.
- * @param columnTextList A list of list of strings for the information to show.
- * @param columnTextSPList A list of list of int that correspond to the size of each string of
- *   columnTextList, (recommend 12.sp for text and 20.sp for number).
- * @param columnTextTagList A list of textTag corresponding to the column.
+ * @param lineTextTag The text tag of the lineText.
+ * @param statsTextList The list of type of stats to display.
+ * @param statsNumberList The list of number for stats to display.
  */
 @Composable
-fun StatisticsRow(
+fun StatisticsColumnRow(
+    columnTestTag: String,
     rowTestTag: String,
     lineText: String,
     lineTextTag: String,
-    columnTextList: List<List<String>>,
-    columnTextSPList: List<List<Int>>,
-    columnTextTagList: List<String>
+    statsTextList: List<String>,
+    statsNumberList: List<String>
 ) {
-  // Ensure that the list have the same size
-  require(
-      columnTextList.size == columnTextSPList.size &&
-          columnTextList.size == columnTextTagList.size) {
-        "All lists must have the same size."
-      }
-  columnTextList.zip(columnTextSPList).forEachIndexed { index, (textList, spList) ->
-    require(textList.size == spList.size) { "Sub lists at index $index must have the same size." }
-  }
-  // Construction of the statistic row
-  Row(
-      modifier = Modifier.fillMaxWidth().padding(vertical = 8.dp).testTag(rowTestTag),
-      horizontalArrangement = Arrangement.SpaceBetween,
-      verticalAlignment = Alignment.CenterVertically) {
+  // Ensure that the lists have the same size
+  require(statsTextList.size == statsNumberList.size) { "The lists must have the same size." }
```

Comment:
This is a pain to deal with. I would find it more intuitive to have a some data class (or simply a `Pair`) group these together. It would help reduce the argument count too.